app/sponsors/controller.js

```diff
@@ -1,29 +1,141 @@
 'use strict';
 
 let Sponsor = require('./model');
+let mv = require('mv');
+let ObjectId = require('mongoose').Types.ObjectId;
+let multiparty = require('multiparty');
+let config = require('../../config/config');
+let path = require('path');
+let projectRoot = require('../../app').projectRoot;
 
 module.exports = {
-  
+
   allSponsors: (req, res) => {
-    Sponsor.find()
+    Sponsor
+    // Exclude the logo because it contains a server path.
+    .find({ }, { logo:0 })
+    .sort({"amount": -1})
     .exec((err, spons) => {
         res.send(spons);
     });
   },
   
   getSponsor: (req, res) => {
-    Sponsor.findById(req.params.id)
+    Sponsor
+    // Exclude the logo because it contains a server path.
+    .findOne({_id: ObjectId(req.params.id)}, { logo:0 })
     .exec((err, spons) => {
         res.send(spons);
     });
   },
 
   getLogo: (req, res) => {
     var id = req.params.id;      
-      res.send(id);
+    Sponsor
+    .findOne({_id: ObjectId(req.params.id)})
+    .select('logo -_id')
+    .exec((err, spons) => {
+
+      // Handle database errors.
+      if (err) {
+        res.sendStatus(500);
+        throw err;
+      }
+      else if (!spons){
+        res.sendStatus(404)
+        return;
+      }
+
+      // Try to send the image if it exists.
+      if (spons.logo) {
+        res.sendFile(spons.logo);
+      } else {
+        res.sendStatus(404);
+      }
+    });
   },
   
   putLogo: (req, res) => {
-    
+    let img = new multiparty.Form();
```

Comment:
I don't really like the put logo process. Best practice is always to not allow users to control (directly at least) the file name that a file gets saved with. As soon as you allow it you have to worry about people manipulating the name it gets saved with. I'm not sure how relevant that is given this is an admin only function, but I get worried easily.

We should probably get the name at least by getting the url param and looking up the sponsor associated with it. I'd prefer just checking that a sponsor with the id exists and then using that for the name.